src/components/Share.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import {shareGoogleDrive} from '../redux/actionCreators';
+import {bindActionCreators} from 'redux';
+import {styles} from './styles';
+import Button from './Button';
+import Spinner from './Spinner';
+import * as selectors from '../redux/selectors';
+
+export class Share extends React.Component {
+  render() {
+    return (
+      <span>
+        {this.props.hasSavedDrive && !this.props.isSharingDrive &&
+          <Button kind="toolbar"
+                  onClick={() => this.props.shareGoogleDrive()}>
+            Share
+          </Button>
+        }
+        {this.props.hasSavedDrive && this.props.isSharingDrive &&
+          <Button kind="toolbar">
+            <Spinner style={styles.spinners.toolbar}/>
+            Sharing...
+          </Button>
+        }
+      </span>
+    );
+  }
+}
+
+Share.propTypes = {
+  hasSavedDrive: React.PropTypes.any,
+  isSharingDrive: React.PropTypes.bool,
+  hasSharedDrive: React.PropTypes.any,
```

Comment:
This is a change I made in the selectors. Before, isSharingDrive() checked the stage of GoogleDrive state, to see if it was 'started', and returned a bool. But, since the stage is overridden throughout the process, we can't rely on that state to determine our current action. Now, the way we check to see if a file has been shared is to check the share state to see if it's empty or not. For now, the share state is a simple string, but I went ahead and used any since I wasn't sure what the state will end up looking like.
I've tried making this something like 
`
export function hasSavedDrive(state) {
  return (state.googleDrive.save === !null);
}
`
But this doesn't seem to work properly. Maybe there's a more elegant way of doing this.